src/devices/Common/Iot/Device/Common/GeographicPosition.cs

```diff
@@ -0,0 +1,371 @@
+            }
+        }
+
+        /// <summary>
+        /// Returns the given angle as degrees, minutes and seconds
+        /// </summary>
+        /// <param name="angle">Input angle, in degrees</param>
+        /// <param name="secDigits">Number of digits for the second</param>
+        /// <param name="normalizedVal">Normalized angle value (to -180 to 180)</param>
+        /// <param name="degrees">Full degrees</param>
+        /// <param name="minutes">Full minutes</param>
+        /// <param name="seconds">Seconds including requested number of digits</param>
+        public static void GetDegreesMinutesSeconds(double angle, int secDigits, out double normalizedVal, out double degrees, out double minutes, out double seconds)
+        {
+            angle = PositionExtensions.NormalizeAngleTo180Degrees(angle);
+            normalizedVal = angle;
+            angle = Math.Abs(angle);
+            degrees = Math.Floor(angle);
+
+            double remains = (angle - degrees) * 60.0;
+            minutes = Math.Floor(remains);
+
+            seconds = (remains - minutes) * 60.0;
+
+            // If rounding the seconds to the given digit would print out "60",
+            // add 1 to the minutes instead
+            if (Math.Round(seconds, secDigits) >= 60.0)
+            {
+                minutes += 1;
+                seconds = 0;
+                if (minutes >= 60) // is basically an integer at this point
+                {
+                    degrees += 1;
+                    minutes = 0;
+                    if (degrees >= 360)
+                    {
+                        degrees -= 360;
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Equality operator. See <see cref="Equals(GeographicPosition?)"/>
+        /// </summary>
+        /// <param name="a">First instance to compare</param>
+        /// <param name="b">Second instance to compare</param>
+        /// <returns>True on equality, false otherwise</returns>
+        public static bool operator ==(GeographicPosition? a, GeographicPosition? b)
+        {
+            if (ReferenceEquals(a, b))
+            {
+                return true;
+            }
+
+            if (ReferenceEquals(a, null))
+            {
+                return false;
+            }
+
+            if (ReferenceEquals(b, null))
+            {
+                return false;
+            }
+
+            return a.Equals(b);
+        }
+
+        /// <summary>
+        /// Inequality operator. See <see cref="Equals(GeographicPosition?)"/>
+        /// </summary>
+        /// <param name="a">First instance to compare</param>
+        /// <param name="b">Second instance to compare</param>
+        /// <returns>True on inequality, false otherwise</returns>
+        public static bool operator !=(GeographicPosition? a, GeographicPosition? b)
+        {
+            return !(a == b);
+        }
+
+        private static string GetEastOrWest(double sign)
```

Comment:
should this be called angle? also might be worth adding a comment about expected range